cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/default-service/AwsDefaults.service.tsx

```diff
@@ -11,4 +11,13 @@ export class AwsDefaults {
     static selectDefalutEC2KeyPairs = (keyPairs: AWSKeyPair[]) => {
         return keyPairs && keyPairs.length > 0 ? keyPairs[0].name : undefined;
     };
+
+    static setDefaultNodeType = (nodeTypeList: string[], nodeProfile: string) => {
+        if (nodeTypeList.indexOf(validateDefaultNodeType(nodeProfile)) > -1) {
+            return validateDefaultNodeType(nodeProfile);
+        } else {
+            // TODO: refactor to select optimal nodeType when preferred default not found
+            return nodeTypeList[nodeTypeList.length / 2];
```

Comment:
javascript can't convert the number to integer automatically. if nodeTypeList.length is an odd number, it will return undefined.

Reply:
Yes sir. Use Math.round(nodeTypeList.length / 2)